01-messaging.md

```diff
@@ -325,24 +327,27 @@ The fundee node:
     - MUST use `temporary_channel_id` in lieu of `channel_id`.
 
 A sending node:
-  - when sending `error`:
-    - MUST fail the channel referred to by the error message.
   - SHOULD send `error` for protocol violations or internal errors that make channels unusable or that make further communication unusable.
   - SHOULD send `error` with the unknown `channel_id` in reply to messages of type `32`-`255` related to unknown channels.
+  - when sending `error`:
+    - MUST fail the channel referred to by the error message.
+  - when sending `warning`:
+	- MAY set `channel_id` to all zero if the warning is not related to a specific channel.
+  - MAY close the connection after sending.
   - MAY send an empty `data` field.
   - when failure was caused by an invalid signature check:
     - SHOULD include the raw, hex-encoded transaction in reply to a `funding_created`, `funding_signed`, `closing_signed`, or `commitment_signed` message.
-  - when `channel_id` is 0:
-    - MUST fail all channels with the receiving node.
-    - MUST close the connection.
-  - MUST set `len` equal to the length of `data`.
 
 The receiving node:
   - upon receiving `error`:
-    - MUST fail the channel referred to by the error message, if that channel is with the sending node.
-  - if no existing channel is referred to by the message:
+    - MUST fail the channel referred to by `channel_id`, if that channel is with the sending node.
+  - upon receiving `warning`:
+    - SHOULD log the message for later diagnosis.
+	- MAY disconnect.
+	- MAY reconnect after some delay to retry.
+	- MAY attempt `shutdown` if permitted at this point.
+  - if no existing channel is referred to by `channel_id`:
```

Comment:
@ariard notes that this appears to apply to warning messages as well, implying that you should ignore and not even bother to log a warning message that is not attached to a channel. I assume that isn't the intent here.

cc https://github.com/rust-bitcoin/rust-lightning/pull/1013#discussion_r683644076

Reply:
I've always treated "ignore" as meaning "don't give a response", not "don't even log".

Reply:
I'm fine with that. I think "ignore" is fine wording.